source/fundamentals/logging.txt

```diff
@@ -0,0 +1,285 @@
+         MongoDatabase database = mongoClient.getDatabase(<my database>);
+         MongoCollection<Document> collection = database.getCollection(<my collection>);
+         collection.find().first();
+
+      You should see output resembling the following:
+
+      .. code-block:: none
+
+         ...
+         1156 [main] DEBUG o.m.driver.protocol.command - Sending command '{"find": "<your collection>", "filter": {}, "limit": 1, "singleBatch": true, "$db": "<your db>", "lsid": {"id": {"$binary": {"base64": "<server session>", "subType": "04"}}}}' with request id 13 to database <your db> on connection <your connection> to server <your server>
+         1173 [main] DEBUG o.m.driver.protocol.command - Execution of command with request id 13 completed successfully in 30.01 ms on connection <your connection> to server <your server>
+   
+      For more information on Logback, see the
+      `Logback manual <http://logback.qos.ch/manual/>`__.
+   
+   .. tab:: Log4j2
+      :tabid: Log4j2-binding
+   
+      If you add these dependencies to your ``pom.xml`` file, the MongoDB Java
+      driver uses the Log4j2 framework to output your logs.
+
+      .. code-block:: xml      
+      
+         <!-- SLF4J API -->
+         <dependency>
+           <groupId>org.slf4j</groupId>
+           <artifactId>slf4j-api</artifactId>
+           <version>1.7.30</version>
+         </dependency>
+         <!-- SLF4J-Log4j2 binding -->
+         <dependency>
+           <groupId>org.apache.logging.log4j</groupId>
+           <artifactId>log4j-slf4j-impl</artifactId>
+           <version>2.14.1</version>
+         </dependency>
+         <!-- Log4j2 core functionality -->
+         <dependency>
+           <groupId>org.apache.logging.log4j</groupId>
+           <artifactId>log4j-core</artifactId>
+           <version>2.14.1</version>
+         </dependency>
+
+      Once you have installed the above dependencies, log an error using the
+      following code.
+
+      .. code-block:: java
+
+         import org.slf4j.Logger;
+         import org.slf4j.LoggerFactory;
+
+         ...
+
+         Logger logger = LoggerFactory.getLogger(App.class);
+         logger.error("Logging an Error");
+
+      You should see output resembling the following:
+
+      .. code-block:: none
+
+         12:35:00.438 [main] ERROR <my package path> - Logging an Error
+
+      For more information on Log4j2, see the
+      `Log4j2 manual <https://logging.apache.org/log4j/2.x/manual/>`__.
+
+Configure Your Logger
+---------------------
+
+To configure your logger, you must configure the logging framework bound to
+SLF4J.
+
+The following subsection provides a concrete example showing how to configure
+your logger.
+
+Configure Your Log Level
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this subsection we show how to configure your logger's **log level**.
+
+A logger's log level specifies a lower bound for how urgent a message must be
+for the logger to output it.
```

Comment:
nitpick: this reads a bit strangely to me, maybe because of `it`.  Perhaps something like
> A logger's log level specified a lower bound for how urgent a message must be to be included in the logger's output.

Reply:
Thanks for the note. My ear does not pick up anything wrong with this sentence, and [the style guide approves](https://docs.mongodb.com/meta/style-guide/style/prepositions/) ending sentences in prepositions, so I am not sure how best to address this feedback. 

My writegood plugin is saying that the suggested change contains passive voice with "to be included" (who is including?). Would changing "it" to "that message" fix the problem you are hearing with the sentence?

Fix: